starknet_py/tests/e2e/client/full_node_test.py

```diff
@@ -38,9 +38,15 @@ async def test_node_get_transaction_by_block_id_and_index(
 
 @pytest.mark.asyncio
 async def test_node_get_deploy_transaction_by_block_id_and_index(
-    deploy_transaction_hash, contract_address, rpc_client, class_hash
+    deploy_transaction_hash,
+    block_with_deploy_number,
+    contract_address,
+    rpc_client,
+    class_hash,
 ):
-    tx = await rpc_client.get_transaction_by_block_id(block_number=1, index=0)
+    tx = await rpc_client.get_transaction_by_block_id(
+        block_number=block_with_deploy_number, index=0
```

Comment:
How are we sure that this transactions will be at index 0 in integration/testnet?

Reply:
Full node tests should not be run in integration/testnet